src/game/Spell.cpp

```diff
@@ -6481,10 +6481,11 @@ bool Spell::CheckTarget(Unit* target, SpellEffectIndex eff) const
             break;
         default:                                            // normal case
             // Get GO cast coordinates if original caster -> GO
-            if (target != m_caster)
-                if (WorldObject* caster = GetCastingObject())
-                    if (!target->IsWithinLOSInMap(caster))
-                        return false;
+            if (!IsIgnoreLosSpell(m_spellInfo))
+                if (target != m_caster)
+                    if (WorldObject* caster = GetCastingObject())
+                        if (!target->IsWithinLOSInMap(caster))
```

Comment:
You are right about assignment, i meant to get rid of a "stair of ifs" into a usual block. This thing justs asks for a mistake to be made at some point. 😄